### DISCUSSION TITLE
Query statistics using query expression

### DISCUSSION THREAD

**Discussion by marcuslindblom** [ravendb/ravendb#12746] [Q&A] [answered]
Is this a valid query using statistics?

```c#
var result = await (from navitem in session.Query<NavigationContext.NavigationItem, NavigationContext>()
    .Statistics(out QueryStatistics statistics)
    where navitem.Parent == currentBlog.Id
    let post = RavenQuery.Load<Post>(navitem.Id)
    select post).ToListAsync();
```

**Comment by ayende** [comment, marked as answer]
Yes, it is. At least at a glance. Did you run into any issues with it?

**Reply by marcuslindblom** [reply]
I don't think so but the value is almost always `-1` but if I restart my app it seems to log a higher value so I guess everything works like it should.

**Reply by ppekrol** [reply]
When DurationInMs is set to -1, it means that the query results came from cache.
